### PR TITLE
Merge selections for same fields (with same alias and arguments) together, as per spec

### LINKED ISSUE #52: Fields are not merged
This query:
```graphql
query { 
  viewer { 
    login
  }
  viewer {
    databaseId
  }
}
```

Should return something like:
```edn
{:data {:viewer {:login "someone" :id "qwertyuiop"}}}
```

Instead, it returns only the `databaseId`. If we query for the login last, it returns just the login.

### PR DESCRIPTION
**Pull request by hlship**
This is my approach to addressing this issue inside the parser; this does not require any execution changes.
 
 I'm going to check the performance impact, if any.

Fixes #52

### CODE REVIEW

**Comment by hlship** [comment]
I can't think of many more possible tests but if you come up with any I'm glad to add them and ensure it all still works.

**Comment by hlship** [comment]
I thought about this on the way home and there's very minor problems with this.  I have ideas for more tests, and for fixes.

**Comment by hlship** [comment]
Happier with this. Collecting performance numbers right now.

File: test/com/walmartlabs/lacinia/merge_selections_test.clj [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+}"))))
+
+(deftest understands-aliases
+  ;; Default for human is 1001 - Darth Vader
+  (is (= {:data {:darth {:name "Darth Vader"
+                         :homePlanet "Tatooine"}}}
+         (q "
+{
+  darth: human { name }
+  darth: human { homePlanet }
```
Comment (bcarrell) [inline test/com/walmartlabs/lacinia/merge_selections_test.clj]: What if these two aliases are different (or one of them don't have an alias).  Are there two resolver calls?

Comment (hlship) [inline test/com/walmartlabs/lacinia/merge_selections_test.clj]: Currently, two resolver calls.  I consider this to be an edge case, and it would be quite disruptive to the executor code to find a way to coalesce this into a single field resolver call.

Comment (bcarrell) [inline test/com/walmartlabs/lacinia/merge_selections_test.clj]: @gusbicalho How do you feel about this case?

Comment (hlship) [inline test/com/walmartlabs/lacinia/merge_selections_test.clj]: Do you know if the RI can do that optimization?

Comment (bcarrell) [inline test/com/walmartlabs/lacinia/merge_selections_test.clj]: Nope, I'm unaware if it can.

**Comment by hlship** [comment]
The CI failure isn't test failing, it's something in the CI server having trouble checking the code out.

File: test/com/walmartlabs/lacinia/merge_selections_test.clj (hunk 1 above)
Comment (gusbicalho) [inline test/com/walmartlabs/lacinia/merge_selections_test.clj]: To me it wouldn't be a problem to make two resolver calls in that case (same field with different aliases), seems to be a really edge case and shouldn't happen in my use case (relay)

**Comment by hlship** [comment]
Yes, there's still some edge cases not covered so well. For example, there's something about fragments defining aliases that are conflicting with the main selection (or perhaps, each other).  There are definitely some parts of the spec that are defined in terms of whats convenient for the RI to implement.
